openvdb/openvdb/tree/ValueAccessor.h

```diff
@@ -604,7 +604,10 @@ class CacheItem
     void addLeaf(LeafNodeType* leaf)
     {
         static_assert(!TreeCacheT::IsConstTree, "can't add a node to a const tree");
-        if (NodeType::LEVEL == 0) return;
+        if (NodeType::LEVEL == 0) {
+            mNext.addLeaf(leaf);
+            return;
+        }
```

Comment:
please change CacheItem::addLeaf to: 

```
void addLeaf(LeafNodeType* leaf)
{
        static_assert(!TreeCacheT::IsConstTree, "can't add a node to a const tree");
        if (NodeType::LEVEL>0 && this->isHashed(leaf->origin())) {
            assert(mNode);
            return const_cast<NodeType*>(mNode)->addLeafAndCache(leaf, *mParent);
        }
        mNext.addLeaf(leaf);
}
```

Reply:
@kmuseth I can verify that OP's changes fix the issues I had recently came across. However, the changes you propose are throwing errors such as

```malloc: *** error for object 0x7f95a2848400: pointer being freed was not allocated.```

Coverting a `Tree5<float, 5, 4, 3, 3>` sphere to a fog volume should be enough to reproduce this.